docs/library/openamp.rst

```diff
@@ -0,0 +1,104 @@
+    rproc.start()
+    
+    while True:
+        if ept.is_ready():
+            ept.send("data")
+
+Functions
+---------
+
+.. function:: ns_callback(ns_callback)
+
+    Set the new service callback.
+
+        - ``ns_callback`` This function will be called back when the remote processor
+          announce new services. At the point the host processor can choose to create the
+          announced endpoint, if this particular service is supported, or ignore it if it's
+          not. If this function is not set, the host processor should first register the
+          endpoint locally, and it will be automatically bound when the remote announces
+          the service.
+
+Classes
+-------
+
+.. class:: Endpoint(name, callback, src_addr=ENDPOINT_ADDR_ANY, dst_addr=ENDPOINT_ADDR_ANY)
+
+   Construct a new RPMsg Endpoint. An endpoint is a bidirectional communication
+   channel between two cores.
+
+        - ``name`` The name of the endpoint.
+        - ``callback`` This function is called when the endpoint receives data with the source
+          address of the remote point, and the data as bytes passed by reference.
+        - ``src_addr`` The endpoint source address. If none is provided, one will be assigned
+          to the endpoint by the library.
+        - ``dst_addr`` The endpoint destination address. This must be provided and must match
+          the remote source address, if the endpoint is create in the ns_callback. Otherwise
+          it will be assigned by the library when the endpoint is bound.
+
+Methods
+-------
+
+.. method:: Endpoint.deinit()
+
+   Destroy the endpoint and release all of its resources.
+
+.. method:: Endpoint.is_ready()
+
+   Returns True if the endpoint is ready to send (i.e., has both a source and destination addresses)
+
+.. method:: Endpoint.send(src_addr=-1, dst_addr=-1, timeout=0)
+
+   Send a message to the remote processor over this endpoint.
+
+        - ``src_addr`` The source endpoint address of the message. If none is provided, the
+          source address the endpoint is bound to is used.
+        - ``dst_addr`` The destination endpoint address of the message. If none is provided,
+          the destination address the endpoint is bound to is used.
+        - ``timeout`` Specifies the time in milliseconds to wait for a free buffer. By default
+          `send()` function is blocking.
+
+.. class:: RemoteProc(entry)
+
+   The RemoteProc object provides processor Life Cycle Management (LCM) support, such as
+   loading firmware, starting and stopping a remote core.
+
+        - ``entry`` Can be a path to firmware image, in that case the firmware is loaded
+          from file to its target memory, or an entry point address, in case the firmware
+          is preloaded or stored on flash.
+
+.. method:: RemoteProc.start()
+
+   Start the remote processor.
+
+.. method:: RemoteProc.stop()
+
+   Stop the remote processor.
+
+.. method:: RemoteProc.shutdown()
```

Comment:
What is the difference between `stop` and `shutdown`?  Can you call `start` after calling `stop`?

Reply:
shutdown stops and releases all of the remote proc's resources. It should also disable power and clocks to the remote core, if possible. Note this is also used as the finaliser.

>>  Can you call start after calling stop?

This is platform dependent. For ST it's Not possible to stop or restart the m4 core, a complete system reset is required.

Reply:
It would be good to add these points to the docs, so users know which of `stop`/`shutdown` they should use and when, and what the restrictions are.